src/test/java/bio/terra/app/controller/StatusTest.java

```diff
@@ -56,36 +71,67 @@ public void testStatusDown() throws Exception {
   }
 
   @Test
-  public void testCriticalSystem() throws Exception {
-    configurationService.setFault(ConfigEnum.CRITICAL_SYSTEM_FAULT.name(), true);
+  public void testCriticalSystemDown() throws Exception {
+    when(datasetDao.statusCheck())
+        .thenReturn(new RepositoryStatusModelSystems().ok(false).critical(true));
     MvcResult result =
         this.mvc.perform(get("/status")).andExpect(status().is5xxServerError()).andReturn();
-    MockHttpServletResponse downResponse = result.getResponse();
-    String responseBody = downResponse.getContentAsString();
+    MockHttpServletResponse response = result.getResponse();
+    String responseBody = response.getContentAsString();
     assertThat(
         "/Status response should indicate that the whole system is down.",
         responseBody,
         startsWith("{\"ok\":false"));
     assertThat(
         "/Status response should indicate that postgres is down",
         responseBody,
-        containsString("\"Postgres\":{\"ok\":false,\"critical\":true"));
+        containsSubserviceString(StatusService.POSTGRES, false, true));
     assertThat(
         "/Status response should indicate that sam is up",
         responseBody,
-        containsString("\"Sam\":{\"ok\":true,\"critical\":true"));
+        containsSubserviceString(StatusService.SAM, true, true));
     assertThat(
         "/Status response should indicate that rbs is up",
         responseBody,
-        containsString("\"ResourceBufferService\":{\"ok\":true,\"critical\":false"));
+        containsSubserviceString(StatusService.RBS, true, false));
+    assertThat(
+        "/Status response should indicate that duos is up",
+        responseBody,
+        containsSubserviceString(StatusService.DUOS, true, false));
+  }
 
-    configurationService.setFault(ConfigEnum.CRITICAL_SYSTEM_FAULT.name(), false);
-    MvcResult upResult = this.mvc.perform(get("/status")).andExpect(status().isOk()).andReturn();
-    MockHttpServletResponse upResponse = upResult.getResponse();
-    String upResponseBody = upResponse.getContentAsString();
+  @Test
+  public void testNonCriticalSystemDown() throws Exception {
+    when(duosService.status())
+        .thenReturn(new RepositoryStatusModelSystems().ok(false).critical(false));
+    MvcResult result = this.mvc.perform(get("/status")).andExpect(status().isOk()).andReturn();
+    MockHttpServletResponse response = result.getResponse();
+    String responseBody = response.getContentAsString();
     assertThat(
-        "/Status response should indicate that the whole system is up",
-        upResponseBody,
+        "/Status response should indicate that the system is up.",
+        responseBody,
         startsWith("{\"ok\":true"));
```

Comment:
Instead of string matching on JSON, I think it would be better to either use `JSONAssert` or `JsonPath`. Both are part of spring boot test so TDR already has them as a test dependency. For example, using `JsonPath` you could write:

```
    assertThat(
        "/Status response should indicate that the system is up.",
        JsonPath.read(responseBody, "$.ok"),
        is("true"));
    assertThat(
        "/Status response should indicate that postgres is up",
        JsonPath.read(responseBody, "$.Postgres.ok"),
        is(true));
```

Reply:
`JsonPath` is built into MvcResult so you could also write
```
    mvc.perform(get("/status")).andExpect(status().isOk())
        .andExpect(jsonPath("$.ok", is("true")))
        .andExpect(jsonPath("$.Postgres.ok", is("true")));
```